libcudacxx/include/cuda/std/detail/libcxx/include/__cccl_config

```diff
@@ -0,0 +1,167 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __CCCL_CONFIG
+#define __CCCL_CONFIG
+
+// Determine the host compiler
+#if defined(__NVCOMPILER)
+#  define _CCCL_COMPILER_NVHPC
+#elif defined(__clang__)
+#  define _CCCL_COMPILER_CLANG
+#elif defined(__GNUC__)
+#  define _CCCL_COMPILER_GCC
+#elif defined(_MSC_VER)
+#  define _CCCL_COMPILER_MSVC
+#elif defined(__IBMCPP__)
+#  define _CCCL_COMPILER_IBM
+#elif defined(__CUDACC_RTC__)
+#  define _CCCL_COMPILER_NVRTC
+#endif
+
+// Convenient shortcut to determine which version of MSVC we are dealing with
+#if defined(_CCCL_COMPILER_MSVC)
+#  if _MSC_VER < 1917
+#    define _CCCL_COMPILER_MSVC_2017
+#  elif _MSC_VER < 1930
+#    define _CCCL_COMPILER_MSVC_2019
+#  else // _MSC_VER < 1940
+#    define _CCCL_COMPILER_MSVC_2022
+#  endif // _MSC_VER < 1940
+#endif // _CCCL_COMPILER_MSVC
+
+// Determine the cuda compiler
+#if defined(__NVCC__)
+#  define _CCCL_CUDA_COMPILER_NVCC
+#elif defined(_NVHPC_CUDA)
+#  define _CCCL_CUDA_COMPILER_NVHPC
+#elif defined(__CUDA__) && defined(_CCCL_COMPILER_CLANG)
+#  define _CCC_CUDA_COMPILER_CLANG
+#endif
+
+// Enforce that cccl headers are treated as system headers
+#if defined(_CCCL_COMPILER_GCC)
+#  define _CCCL_FORCE_SYSTEM_HEADER _Pragma("GCC system_header")
+#elif defined(_CCCL_COMPILER_CLANG)
+#  define _CCCL_FORCE_SYSTEM_HEADER _Pragma("clang system_header")
+#elif defined(_CCCL_COMPILER_MSVC)
+#  define _CCCL_FORCE_SYSTEM_HEADER __pragma(system_header)
+#elif defined(_CCCL_COMPILER_NVHPC)
+#  define _CCCL_FORCE_SYSTEM_HEADER _Pragma("GCC system_header")
+#else
+#  define _CCCL_FORCE_SYSTEM_HEADER
+#endif
+
+// We want to ensure that all warning emmiting from this header are supressed
+_CCCL_FORCE_SYSTEM_HEADER
```

Comment:
This appears to be necessary otherwise the visibility attribute will explode on us 🤷‍♂️